libevmcore/Assembly.cpp

```diff
@@ -156,6 +163,133 @@ ostream& Assembly::stream(ostream& _out, string const& _prefix, StringMap const&
 	return _out;
 }
 
+Json::Value Assembly::createJsonValue(string _name, int _locationX, int _locationY, string _value, string _jumpType) const
+{
+	Json::Value value;
+	value["name"] = _name;
+	value["begin"] = _locationX;
+	value["end"] = _locationY;
+	if (!_value.empty())
+		value["value"] = _value;
+	if (!_jumpType.empty())
+		value["jumpType"] = _jumpType;
+	return value;
+}
+
+Json::Value Assembly::streamAsmJson(ostream& _out, string const& _prefix, StringMap const& _sourceCodes, bool _inJsonFormat) const
+{
+	Json::Value root;
+
+	string currentArrayName = ".code";
+	std::vector<Json::Value> currentCollection;
+	for (AssemblyItem const& i: m_items)
```

Comment:
`m_items` has to be transformed into a single JSON array and not a JSON hashtable. The reason is that we can have tags/JUMPDESTS that are not only jumped to, but also "overflown into", i.e. control flow just continues into a JUMPDEST.
I think the best solution would be to treat a tag just as a special type of instruction, as we also do with PUSHSUBSIZE and others.